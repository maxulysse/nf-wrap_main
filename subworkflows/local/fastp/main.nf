// FASTP

include { FASTP } from '../../../modules/nf-core/fastp/main.nf'

workflow RUN_FASTP {
    take:
    input             // channel: [mandatory] [ meta, file.fastq.gz ] or [meta, [ file_1.fastq.gz, file_2.fastq.gz ] ]
    adapter           // channel: [optional] [ adapter_fasta ]
    save_trimmed_fail // boolean: [optional] save trimmed fail files
    save_merged       // boolean: [optional] save merged files

    main:
    FASTP(input, adapter, save_trimmed_fail, save_merged)

    emit:
    html         = FASTP.out.html
    json         = FASTP.out.json
    logs         = FASTP.out.log
    reads        = FASTP.out.reads
    reads_fail   = FASTP.out.reads_fail
    reads_merged = FASTP.out.reads_merged
    versions     = FASTP.out.versions
}
