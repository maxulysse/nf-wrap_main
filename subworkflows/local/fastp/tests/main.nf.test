nextflow_workflow {

    name "Test Workflow RUN_FASTP"
    script "../main.nf"
    workflow "RUN_FASTP"
    tag "nf-wrap/fastp"

    test("FASTP") {

        when {
            params {
            }
            workflow {
                """
                adapter_fasta     = []
                save_trimmed_fail = false
                save_merged       = false

                input[0] = Channel.of([
                    [ id:'test', single_end:true ], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true)
                ])
                input[1] = adapter_fasta
                input[2] = save_trimmed_fail
                input[3] = save_merged
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(file(workflow.out.html[0][1]).name).match("html") },
                { assert snapshot(file(workflow.out.logs[0][1]).name).match("logs") },
                { assert snapshot(workflow.out.json).match("json")},
                { assert snapshot(workflow.out.reads).match("reads")},
                { assert snapshot(workflow.out.reads_fail).match("reads_fail")},
                { assert snapshot(workflow.out.reads_merged).match("reads_merged")},
                { assert snapshot(workflow.out.versions).match("versions")}
            )
        }
    }
}
