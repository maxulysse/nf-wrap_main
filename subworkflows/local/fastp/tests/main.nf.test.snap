{
    "versions": {
        "content": [
            [
                "versions.yml:md5,7c77c31dcfc021f8d85d267d8b8936df"
            ]
        ],
        "timestamp": "2023-11-25T11:50:40.915718"
    },
    "reads": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test.fastp.fastq.gz:md5,dc08e42319d0deab088da4d8600aa2bd"
                ]
            ]
        ],
        "timestamp": "2023-11-25T11:50:40.903104"
    },
    "json": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test.fastp.json:md5,c852d7a6dba5819e4ac8d9673bedcacc"
                ]
            ]
        ],
        "timestamp": "2023-11-25T11:50:40.899211"
    },
    "html": {
        "content": [
            "test.fastp.html"
        ],
        "timestamp": "2023-11-25T11:50:40.890972"
    },
    "logs": {
        "content": [
            "test.fastp.log"
        ],
        "timestamp": "2023-11-25T11:50:40.897481"
    },
    "reads_fail": {
        "content": [
            [
                
            ]
        ],
        "timestamp": "2023-11-25T11:50:40.907646"
    },
    "reads_merged": {
        "content": [
            [
                
            ]
        ],
        "timestamp": "2023-11-25T11:50:40.911819"
    }
}